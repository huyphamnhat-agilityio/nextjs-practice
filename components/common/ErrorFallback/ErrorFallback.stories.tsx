import type { Meta, StoryObj } from "@storybook/react";

// Components
import { ErrorFallback } from "@/components";

const meta: Meta<typeof ErrorFallback> = {
  title: "Components/common/ErrorFallback",
  component: ErrorFallback,
  argTypes: {
    message: {
      description: "The error message to display",
    },
  },
  args: {},
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof ErrorFallback>;

export const Primary: Story = {};

export const WithMessage: Story = {
  args: {
    message: "Sample error message",
  },
};
