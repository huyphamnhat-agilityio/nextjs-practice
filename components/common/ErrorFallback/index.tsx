import Link from "next/link";

// Components
import { ErrorIcon } from "@/components";
import { DESTINATION } from "@/constants";

export type ErrorFallbackProps = {
  message?: string;
};

const ErrorFallback = ({ message = "" }: ErrorFallbackProps) => {
  return (
    <div className="flex flex-col items-center gap-6 max-w-xs sm:max-w-full">
      <ErrorIcon />
      <h4 className="text-foreground text-4xl text-center">
        Oops! There is something wrong
      </h4>
      <p className="text-foreground-100 text-xl text-center whitespace-pre-line">
        {`An error occurred. For more help, feel free 
        to reach out to our support team.`}
      </p>

      {message && (
        <p className="text-foreground-100 text-lg text-center whitespace-pre-line">
          Detail error:
          <br />
          <span className="text-danger">{message}</span>
        </p>
      )}

      <p className="text-foreground text-xl text-center">
        Please refresh the page or{" "}
        <Link href={DESTINATION.HOME} className="text-primary">
          back to home
        </Link>
      </p>
    </div>
  );
};

export default ErrorFallback;
