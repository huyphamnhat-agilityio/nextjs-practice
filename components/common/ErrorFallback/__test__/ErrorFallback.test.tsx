import { render } from "@testing-library/react";

import ErrorFallback, { ErrorFallbackProps } from "..";

describe("ErrorFallback test cases", () => {
  const setup = (props: ErrorFallbackProps) =>
    render(<ErrorFallback {...props} />);

  it("should render correctly", () => {
    const { asFragment } = setup({});

    expect(asFragment()).toMatchSnapshot();
  });
});
