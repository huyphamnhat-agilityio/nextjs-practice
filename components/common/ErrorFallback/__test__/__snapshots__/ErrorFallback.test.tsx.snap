// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFallback test cases should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center gap-6 max-w-xs sm:max-w-full lg:min-h-[66vh]"
  >
    <svg
      fill="none"
      height="200"
      viewBox="0 0 365 258"
      width="300"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M216.125 64.2221C218.749 53.3617 234.251 53.3617 236.875 64.2221C240.111 77.6138 255.386 84.1394 267.281 77.1179L270.29 75.3414C279.153 70.1095 289.145 80.567 283.273 89.2598C275.466 100.817 281.817 116.651 295.492 119.539C305.721 121.699 305.721 136.301 295.492 138.461C281.817 141.349 275.466 157.183 283.273 168.74C289.145 177.433 279.153 187.89 270.29 182.659L267.281 180.882C255.386 173.861 240.111 180.386 236.875 193.778C234.251 204.638 218.749 204.638 216.125 193.778C212.889 180.386 197.614 173.861 185.719 180.882L182.71 182.659C173.847 187.89 163.855 177.433 169.727 168.74C177.534 157.183 171.183 141.349 157.508 138.461C147.279 136.301 147.279 121.699 157.508 119.539C171.183 116.651 177.534 100.817 169.727 89.2598C163.855 80.567 173.847 70.1095 182.71 75.3414L185.719 77.1179C197.614 84.1394 212.889 77.6138 216.125 64.2221Z"
        stroke="#FF6551"
        stroke-width="10"
      />
      <path
        d="M256.311 128.999C256.311 143.542 243.417 155.998 226.656 155.998C209.894 155.998 197 143.542 197 128.999C197 114.456 209.894 102 226.656 102C243.417 102 256.311 114.456 256.311 128.999Z"
        stroke="#FF6551"
        stroke-width="10"
      />
      <path
        d="M84.0695 133.742C85.3069 128.604 92.6489 128.604 93.8862 133.742C96.2392 143.514 107.39 148.276 116.067 143.136L117.897 142.052C121.929 139.663 126.597 144.413 123.874 148.458C118.167 156.935 122.823 168.534 132.848 170.659C137.538 171.653 137.538 178.347 132.848 179.341C122.823 181.466 118.167 193.065 123.874 201.542C126.597 205.587 121.929 210.337 117.897 207.948L116.067 206.864C107.39 201.724 96.2392 206.486 93.8862 216.258C92.6489 221.396 85.3069 221.396 84.0695 216.258C81.7165 206.486 70.5658 201.724 61.8891 206.864L64.4374 211.166L61.8891 206.864L60.0589 207.948C56.0266 210.337 51.3592 205.587 54.0821 201.542C59.7888 193.065 55.1332 181.466 45.1078 179.341C40.4173 178.347 40.4173 171.653 45.1078 170.659C55.1332 168.534 59.7888 156.935 54.0821 148.458C51.3592 144.413 56.0266 139.663 60.0589 142.052L61.8891 143.136C70.5658 148.276 81.7165 143.514 84.0695 133.742Z"
        stroke="#FF6551"
        stroke-width="10"
      />
      <path
        d="M104 175C104 181.697 97.7928 188 89 188C80.2072 188 74 181.697 74 175C74 168.303 80.2072 162 89 162C97.7928 162 104 168.303 104 175Z"
        stroke="#FF6551"
        stroke-width="10"
      />
    </svg>
    <h4
      class="text-foreground text-4xl text-center"
    >
      Oops! There is something wrong
    </h4>
    <p
      class="text-foreground-100 text-xl text-center whitespace-pre-line"
    >
      An error occurred. For more help, feel free 
        to reach out to our support team.
    </p>
    <p
      class="text-foreground text-xl text-center"
    >
      Please refresh the page or 
      <a
        class="text-primary"
        href="/"
      >
        back to home
      </a>
    </p>
  </div>
</DocumentFragment>
`;
